### PR TITLE
Fixes #4 create list contact UI

### LINKED ISSUE #4: Create list contacts UI

### PR DESCRIPTION
**Pull request by divinsmathew**

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Contacts/ContactList.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            user={{
+              name: data.firstName + " " + data.secondName,
+              imageUrl: data.profilePicUrl,
+            }}
+            size="medium"
+            className="mr-3"
+          />
+          <div>
+            <Typography style="h5" weight="semibold">
+              {data.firstName + " " + data.secondName}
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/ContactList.jsx:28]: Let's use template strings here.

File: app/javascript/src/components/Dashboard/Contacts/ContactList.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  const COLUMN_DATA = [
+    {
+      title: "NAME & ROLE",
+      dataIndex: "nameAndRole",
+      key: "nameAndRole",
+      render: (_, data) => (
+        <div className="flex">
+          <Avatar
+            user={{
+              name: data.firstName + " " + data.secondName,
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/ContactList.jsx:20]: Let's use template strings here.

File: app/javascript/src/components/Dashboard/Contacts/ContactList.jsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import React from "react";
+
+import { MenuHorizontal } from "@bigbinary/neeto-icons";
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/ContactList.jsx:3]: Please use aliases.

File: app/javascript/src/components/Dashboard/Contacts/ContactList.jsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import { Typography, Avatar, Dropdown, Table, Pagination } from "neetoui/v2";
+
+export default function ContactList({
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/ContactList.jsx:6]: `const ContactList = () => {}`

File: app/javascript/src/components/Dashboard/Contacts/ContactList.jsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        columnData={COLUMN_DATA}
+        currentPageNumber={3}
+        totalCount={10}
+      />
+      <Pagination
+        count={300}
+        pageNo={3}
+        pageSize={25}
+        className="flex justify-end mt-9"
+      />
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/ContactList.jsx:83]: We don't need this anymore. `Table` component auto-generates pagination.

File: app/javascript/src/components/Dashboard/Contacts/ContactsMenu.jsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import React from "react";
+
+import { Search, Settings, Plus } from "@bigbinary/neeto-icons";
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/ContactsMenu.jsx:3]: Please use aliases.

File: app/javascript/src/components/Dashboard/Contacts/ContactsMenu.jsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import { Typography } from "neetoui/v2";
+import { MenuBar } from "neetoui/v2/layouts";
+
+const ContactsMenu = ({ showContactsMenu, all }) => {
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/ContactsMenu.jsx:7]: We don't need to suffix files/folders inside of `Contacts` with `Contact`.

File: app/javascript/src/components/Dashboard/Contacts/DeleteAlert.jsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+      onSubmit={handleDelete}
+      onClose={onClose}
+      message="Are you sure you want to delete this contact? This cannot be undone."
+      title={`Delete Contact`}
+      isSubmitting={deleting}
+    />
+  );
+};
+
+export default DeleteAlert;
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/DeleteAlert.jsx:34]: Can we make this into a reusable component for both contacts and notes?

File: app/javascript/src/components/Dashboard/Contacts/constants.jsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
+export const dummyContacts = [
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/constants.jsx:1]: `export const DUMMY_CONTACTS = [`

File: app/javascript/src/components/Dashboard/Contacts/index.jsx [hunk 10]
```diff
@@ -0,0 +1,71 @@
+import React, { useState } from "react";
+
+import { Plus } from "@bigbinary/neeto-icons";
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/index.jsx:3]: Please use aliases.

File: app/javascript/src/components/Dashboard/Contacts/index.jsx [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+import EmptyState from "components/Common/EmptyState";
+
+import { dummyContacts } from "./constants";
+import ContactList from "./ContactList";
+import ContactsMenu from "./ContactsMenu";
+import DeleteAlert from "./DeleteAlert";
+
+function Contacts() {
+  const [showContactsMenu, setShowContactsMenu] = useState(true);
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/index.jsx:16]: `const [isMenuOpen, setIsMenuOpen] = useState(true);`

File: app/javascript/src/components/Dashboard/Contacts/index.jsx [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+  const [selectedContactId, setSelectedContactId] = useState(false);
+  const [contacts, setContacts] = useState(dummyContacts);
+
+  return (
+    <section className="flex w-full">
+      <section>
+        <ContactsMenu
+          all={contacts.length}
+          showContactsMenu={showContactsMenu}
+        ></ContactsMenu>
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/index.jsx:27]: Can't we use self closing tags here?

File: app/javascript/src/components/Dashboard/Contacts/index.jsx [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+            <>
+              <Input
+                className="mr-4 w-72"
+                placeholder="Search Name, Email, Phone Number, Ect."
+              />
+              <Button label="New Contact" icon={Plus} />
+            </>
+          }
+        />
+        {contacts.length ? (
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/index.jsx:43]: Please use ramdajs here.

File: app/javascript/src/components/Dashboard/Contacts/index.jsx [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+import EmptyState from "components/Common/EmptyState";
+
+import { dummyContacts } from "./constants";
+import ContactList from "./ContactList";
+import ContactsMenu from "./ContactsMenu";
+import DeleteAlert from "./DeleteAlert";
+
+function Contacts() {
+  const [showContactsMenu, setShowContactsMenu] = useState(true);
+  const [showDeleteAlert, setShowDeleteAlert] = useState(false);
```
Comment (karthiknmenon) [inline app/javascript/src/components/Dashboard/Contacts/index.jsx:17]: `const [isDeleteAlertOpen, setIsDeleteAlertOpen] = useState(false);`

File: tailwind.config.js [hunk 15]
```diff
@@ -16,6 +16,9 @@ module.exports = {
     defaultExtractor: content => content.match(/[A-Za-z0-9-_:/]+/g) || [],
   },
   theme: {
+    maxWidth: {
+      'full': '99%',
```
Comment (karthiknmenon) [inline tailwind.config.js:20]: Why do we need this?

**Review by karthiknmenon — COMMENTED** [review]
@divinsmathew Please check the suggestions.

File: tailwind.config.js (hunk 15 above)
Comment (divinsmathew) [reply tailwind.config.js:20]: Was going for a temporary hack to fix width issue. Removed it and fixed it properly now.

**Pull request merged**